src/lib.rs

```diff
@@ -62,69 +106,466 @@ fn validate(payload: &[u8]) -> CallResult {
 
-    use kubewarden_policy_sdk::test::Testcase;
+        // needed for creating mocks
+        #[allow(dead_code)]
+        pub fn verify_pub_keys_image(
+            _image: &str,
+            _pub_keys: Vec<String>,
+            _annotations: Option<HashMap<String, String>>,
+        ) -> Result<VerificationResponse> {
+            Ok(VerificationResponse {
+                is_trusted: true,
+                digest: "mock_digest".to_string(),
+            })
+        }
+
+        // needed for creating mocks
+        #[allow(dead_code)]
+        pub fn verify_keyless_exact_match(
+            _image: &str,
+            _keyless: Vec<KeylessInfo>,
+            _annotations: Option<HashMap<String, String>>,
+        ) -> Result<VerificationResponse> {
+            Ok(VerificationResponse {
+                is_trusted: true,
+                digest: "mock_digest".to_string(),
+            })
+        }
+    }
 
+    // these tests need to run sequentially because mockall creates a global context to create the mocks
     #[test]
-    fn accept_pod_with_valid_name() -> Result<(), ()> {
-        let request_file = "test_data/pod_creation.json";
+    #[serial]
+    fn pub_keys_validation_pass_with_mutation() {
+        let ctx = mock_sdk::verify_pub_keys_image_context();
+        ctx.expect().times(1).returning(|_, _, _| {
+            Ok(VerificationResponse {
+                is_trusted: true,
+                digest: "sha256:89102e348749bb17a6a651a4b2a17420e1a66d2a44a675b981973d49a5af3a5e"
+                    .to_string(),
+            })
+        });
+
+        let settings: Settings = Settings {
+            signatures: vec![Signature::PubKeys {
+                0: PubKeys {
+                    image: "nginx:*".to_string(),
+                    pub_keys: vec!["key".to_string()],
+                    annotations: None,
+                },
+            }],
+            modify_images_with_digest: true,
+        };
+
         let tc = Testcase {
-            name: String::from("Valid name"),
-            fixture_file: String::from(request_file),
+            name: String::from("It should successfully validate the nginx container"),
+            fixture_file: String::from("test_data/pod_creation.json"),
+            settings: settings,
             expected_validation_result: true,
-            settings: Settings {},
         };
 
-        let res = tc.eval(validate).unwrap();
-        assert!(
-            res.mutated_object.is_none(),
-            "Something mutated with test case: {}",
-            tc.name,
-        );
+        let response = tc.eval(validate).unwrap();
+        assert_eq!(response.accepted, true);
+        let expected_mutation: serde_json::Value = serde_json::from_str("{\"apiVersion\":\"v1\",\"kind\":\"Pod\",\"metadata\":{\"name\":\"nginx\"},\"spec\":{\"containers\":[{\"image\":\"nginx:v1.26@sha256:89102e348749bb17a6a651a4b2a17420e1a66d2a44a675b981973d49a5af3a5e\",\"name\":\"nginx\"}]}}").unwrap();
```

Comment:
There's a nicer way to declare that, `serde_json` provides a `json!` macro that can be used like that:

```rust
let my_serde_json_value: serde_json::Value = json!({
  "hello": "world",
  "times": 2,
});
```